__tests__/reducers/literal.test.js

```diff
@@ -1,88 +1,136 @@
-import { literal } from '../../src/reducers/literal'
+import { removeAllContent, removeMyItem, setMyItems  } from '../../src/reducers/literal'
 
 describe('literal reducer', () => {
-  it('should handle initial state', () => {
-    expect(
-      literal(undefined, {})
-    ).toEqual({formData: []})
-  })
 
   it('should handle SET_ITEMS', () => {
-    expect(
-      literal({formData: []}, {
-        type: 'SET_ITEMS',
-        payload: {id: 1, uri:'Run the tests', items: []}
-      })
-    ).toEqual({
-      "formData": [{
-        "id": 1, "uri": "Run the tests", "items": []
-      }]
+    const literalSetItems = setMyItems({ "resourceTemplate:Monograph:Instance": {
+      'http://schema.org/name': { items: [] }
+    }}, {
+      type: 'SET_ITEMS',
+      payload: {
+        rtId: 'resourceTemplate:Monograph:Instance',
+        uri:'http://schema.org/name',
+        items: [ { id: 0, content: 'Run the tests'} ]
+      }
+    })
+    expect(literalSetItems).toEqual({
+      "resourceTemplate:Monograph:Instance": {
+        'http://schema.org/name': {
+          items: [{ id: 0, content: 'Run the tests'}]
+        }
+      }
```

Comment:
I'm a bit muzzy on the diff between the old and the new, but the OLD way showed items as an empty array in all cases, whereas this new one shows items populated ...  in both cases, I'm not clear on what SET_ITEMS is supposed to be doing.  "handle SET_ITEMS" is pretty vague as a description.

I note the new test code certainly seems no worse than the old, in that regard ...

Reply:
Storing the InputLiterals state in the Redux is now changed to support selector pattern; changed the test description from the vague to hopefully a little more specific on what calling **SET_ITEMS** does to the Redux state for a specific InputLiteral.